internal/v1/handler_blueprints.go

```diff
@@ -176,6 +178,12 @@ func (h *Handlers) UpdateBlueprint(ctx echo.Context, blueprintId uuid.UUID) erro
 }
 
 func (h *Handlers) ComposeBlueprint(ctx echo.Context, id openapi_types.UUID) error {
+	var requestBody ComposeBlueprintJSONBody
+	err := ctx.Bind(&requestBody)
+	if err != nil {
+		return err
+	}
```

Comment:
Just to be sure, these 5 lines are breaking tests. The loading (binding) should be optional. I am not sure how to do this with echo library tho, I have never done that. The docs say that binding depends on Content-Type header. Perhaps check the error when you call the endpoint without any body:

```
POST .../blueprints/UUID/compose
Content-Length: 0
```

or

```
POST .../blueprints/UUID/compose
Content-Length: 0
Content-Type: application/json
```

In both cases, I think the endpoint should work as before - compose all image types. See what UI does in Chrome inspector.

Reply:
Yeah [here is the code of Bind](https://github.com/labstack/echo/blob/master/bind.go#L67).

* It returns immediately if `Content-Length: 0` this is what should be set by UI and tests at the moment.
* It tries to unmarshal JSON if and only if `Content-Length` is higher than zero and `Content-Type` is `application/json`.

Reply:
@jrusz can you tell what `Content-Length` and `Content-Type` headers tests are passing at the moment?